### PR TITLE
Fix character offset off-by-one error in enhanced line directives

### PR DESCRIPTION
**Pull request by jjonescz**
Fixes https://github.com/dotnet/razor/issues/9051.

The character offset should be "number of characters" per [the spec](https://github.com/dotnet/csharplang/blob/b84e766ec04eccd8d74fbcf01815fde671bc09bf/proposals/csharp-10.0/enhanced-line-directives.md#detailed-design). It was internally represented in `LineMappingEntry.UnmappedCharacterOffset` as "number of characters - 1", causing off-by-one errors (those could probably be fixed also by leaving the representation and adding some +-1 where necessary, but this seems cleaner).

### CODE REVIEW

**Comment by jjonescz** [comment]
@tmat @dotnet/roslyn-compiler for reviews, thanks

**Review by cston — APPROVED** [review]

File: src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public enum PositionState : byte
             // 0-based mapped span from enhanced #line directive.
             public readonly LinePositionSpan MappedSpan;
 
-            // optional 0-based character offset from enhanced #line directive.
+            // optional character offset from enhanced #line directive.
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs:69]: Consider keeping a mention of the base ("1-based ...")

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3) with a nit to consider

File: src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs (hunk 1 above)
Comment (jjonescz) [reply src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs:69]:
```suggestion
            // optional 1-based character offset from enhanced #line directive.
```

Comment (jjonescz) [reply src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs:69]: It doesn't really have a base, I think - it's just a number of characters - 0 means 0 characters, 1 means 1 character, ...

**Pull request merged**